checks.d/storm_rest_api.py

```diff
@@ -0,0 +1,301 @@
+
+        details = None
+        with open(config.cache_file, 'r') as cache_f:
+            details = json.load(cache_f)
+
+        oldest_acceptable_cache_time = time.time() - config.cache_staleness
+        if details['status'] == 'success' and details['updated'] > oldest_acceptable_cache_time:
+            data = details['data']
+            self.report_cluster(config, data['cluster'])
+            self.report_supervisors(config, data['supervisors'])
+            self.report_topologies(config, data['topologies'])
+            for topology_detail in data['topology_details']:
+                self.report_topology(config, topology_detail['name'], topology_detail['topology'])
+                for executor_details in topology_detail['component_details']:
+                    self.report_executor_details(config, executor_details)
+        else:
+            check_status = AgentCheck.CRITICAL
+            if details['status'] != 'success':
+                check_msg = "Could not connect to URL %s with timeout %d" % (failure.url, failure.timeout)
+            else:
+                check_msg = "Cache is too stale: %d vs. expected minimum %d" % (details['updated'], oldest_acceptable_cache_time)
+
+        self.service_check(
+            self.metric(config, 'storm.rest.cached_data_ok'),
+            check_status,
+            message=check_msg,
+            tags=check_tags)
+
+    def report_cluster(self, config, cluster):
+        self.monotonic_count(self.metric(config, 'storm.rest.cluster.nimbus_uptime_seconds'),
+                             storm_utils.translate_timespec(cluster['nimbusUptime']),
+                             tags=config.tags)
+        self.gauge(self.metric(config, 'storm.rest.cluster.slots_used_count'),
+                   cluster['slotsUsed'],
+                   tags=config.tags)
+        self.gauge(self.metric(config, 'storm.rest.cluster.supervisor_count'),
+                   cluster['supervisors'],
+                   tags=config.tags)
+        self.gauge(self.metric(config, 'storm.rest.cluster.slots_total_count'),
+                   cluster['slotsTotal'],
+                   tags=config.tags)
+        self.gauge(self.metric(config, 'storm.rest.cluster.slots_free_count'),
+                   cluster['slotsFree'],
+                   tags=config.tags)
+        self.gauge(self.metric(config, 'storm.rest.cluster.executors_total_count'),
+                   cluster['executorsTotal'],
+                   tags=config.tags)
+        self.gauge(self.metric(config, 'storm.rest.cluster.tasks_total_count'),
+                   cluster['tasksTotal'],
+                   tags=config.tags)
+
+
+    def report_supervisors(self, config, resp):
+        self.gauge(self.metric(config, 'storm.rest.supervisors_total'),
+                   len(resp.get('supervisors', [])),
+                   tags=config.tags)
+        for supe in resp.get('supervisors', []):
+            supe_tags = [
+                'host:' + supe.get('host'),
```

Comment:
Not sure if this is the intention, but I believe this will overwrite the normal `host` metric that covers where these metrics came from. I'm not sure where I saw that, as I can't find it now…